### DISCUSSION TITLE
ImageRepository reconciliation error

### DISCUSSION THREAD

**Discussion by Emilius1** [fluxcd/flux2#3495] [Q&A] [answered]
Hello Community,
For unknown reason to me my auto image tag update stopped working. It seems Image Reflector Controller doesn't get the latest image tags from my image repo.
I've tried with >flux reconcile image repository <my_ImageRepository> but I receive "client rate limiter Wait returned an error: context deadline exceeded"
And I'm lost. Any hint what to check/do?
Thanks

**Comment by kingdonb** [comment]
You should check the `describe` output of the `ImageRepository`, at the bottom there is a list of events, and in the YAML you can also check the `status` for `conditions` which might give some further information about what's going wrong.

The `client rate limiter Wait returned an error: context deadline exceeded` seems like not a very helpful error message. Mind telling us what you find in the status, and we can see if the `flux` CLI should have given you better feedback? I think that error isn't helpful.

**Comment by Emilius1** [comment, marked as answer]
Hi @kingdonb,
I've found the error. The reason was running again flux bootstrap, but forgetting to add params:
`--components-extra=image-reflector-controller,image-automation-controller`
Which was removing the needed controllers/pods. (image-reflector(automatio)-controller)
Yet, `describe`  command didn't help to understand the error.
As well `flux get image repository <name>` was returning that everything is normal.

**Reply by kingdonb** [reply]
That is surprising because if the extra components are not installed, then their CRDs should neither be. I would expect `flux get image repository ...` to return an error, if the image repository did not exist or the kinds were not on the cluster connected.

Anyway, I'm glad you solved it! If you found any usability issue that you think needs follow-up, please do not hesitate to submit an issue, we'll follow it up. Thanks for using Flux!
